crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -534,6 +543,157 @@ impl kitsune2_api::KitsuneHandler for HolochainP2pActor {
             }
         })
     }
+
+    fn preflight_gather_outgoing(&self, _peer_url: Url) -> K2Result<bytes::Bytes> {
+        Ok(self.preflight.lock().unwrap().clone())
+    }
+
+    fn preflight_validate_incoming(&self, _peer_url: Url, data: bytes::Bytes) -> K2Result<()> {
+        // decode the preflight that the remote sent us
+        let rem = crate::wire::WirePreflightMessage::decode(&data)
+            .map_err(|err| K2Error::other_src("Invalid remote preflight", err))?;
+
+        // if the compats don't match, reject the connection
+        if rem.compat != self.compat {
+            return Err(K2Error::other(format!(
+                "Invalid remote preflight, wanted {:?}, got: {:?}",
+                self.compat, rem.compat
+            )));
+        }
+
+        let mut agents = Vec::new();
+
+        // decode the agents inline, so we can reject the connection
+        // if they sent us bad agent data
+        for agent in rem.agents {
+            agents.push(AgentInfoSigned::decode(
+                &kitsune2_core::Ed25519Verifier,
+                agent.as_bytes(),
+            )?);
+        }
+
+        // if they sent us agents, spawn a task to insert them into the store
+        if !agents.is_empty() {
+            let kitsune = self.kitsune.clone();
+            tokio::task::spawn(async move {
+                for agent in agents {
+                    let space = match kitsune.space_if_exists(agent.space.clone()).await {
```

Comment:
Why would they send us agents of a different space? Given they're not malicious.
I'd expect only agents of this space. In that case the agents could be filtered by this space.

Reply:
Remember connections are shared among spaces, so there is a single preflight for all spaces. There is no "this" space. Moreover, the remote could have cells in dnas that are not even installed on this node. We don't want to create those spaces here just to put an agent info in it that will be ignore. We want to ignore that agent info.

Reply:
Ah, this is a connection. Yes, thanks for the explanation. So we're inserting all agents if we're part also in their space. Got it.